src/main/java/org/mskcc/cbio/oncokb/service/UserService.java

```diff
@@ -646,17 +659,41 @@ public Optional<UserDTO> updateUserWithCompanyLicense(UserDTO userDTO, Company c
      * Deactivates the user and also expires all their tokens.
      * @param userDTO the userDTO
      */
-    private void expireUserAccount(UserDTO userDTO){
+    private void expireUserAccount(UserDTO userDTO) {
         userDTO.setActivated(false);
-        Optional<UserDTO> updatedUserDTO = updateUser(userDTO);
-        if(updatedUserDTO.isPresent()){
+        Optional<UserDTO> optionalUpdatedUserDTO = updateUser(userDTO);
+        if (optionalUpdatedUserDTO.isPresent()) {
+            UserDTO updatedUserDTO = optionalUpdatedUserDTO.get();
+            if (userHasUnactivatedTrial(updatedUserDTO)) {
+                clearTrialAccountInformation(updatedUserDTO);
+            }
             List<Token> tokens = tokenService.findByUser(userMapper.userDTOToUser(userDTO));
             tokens.forEach(token -> {
                 tokenService.expireToken(token);
             });
         }
     }
 
+    /**
+     * Clears the trial account information. If clearing the trial account information makes the additional
+     * information empty, then additional info is set to null.
+     * @param userDTO
+     */
+    private void clearTrialAccountInformation(UserDTO userDTO) {
+            userDTO.getAdditionalInfo().setTrialAccount(null);
+            Optional<UserDetails> userDetails = userDetailsRepository.findOneByUser(userMapper.userDTOToUser(userDTO));
+            if (userDetails.isPresent()) {
+                UserDetails ud = userDetails.get();
+                String newAdditionalInfoString = null;
+                // Preserve userCompany information if present
+                if (userDTO.getAdditionalInfo().getUserCompany() != null) {
+                    newAdditionalInfoString = new Gson().toJson(userDTO.getAdditionalInfo());
+                }
+                ud.setAdditionalInfo(newAdditionalInfoString);
```

Comment:
UserCompany is not guaranteed to be the only data element besides the trial account in the additional info in the future. I would just leave the additional info there.

Reply:
I removed the code for checking `userCompany` and just set the `trialAccount` to `null`. Then I added a util function to check if the `additionalInfo` object is empty.